document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFunctionExpressionVisitor.java

```diff
@@ -33,21 +47,19 @@ public String visit(final FunctionExpression expression) {
           String.format("%s should have at least one operand", expression));
     }
 
-    PostgresSelectTypeExpressionVisitor selectTypeExpressionVisitor =
-        new PostgresDataAccessorIdentifierExpressionVisitor(
-            new PostgresConstantExpressionVisitor(getPostgresQueryParser()));
-
     if (numArgs == 1) {
-      String value = expression.getOperands().get(0).accept(selectTypeExpressionVisitor);
-      return String.format("%s( %s )", expression.getOperator().toString(), value);
+      String parsedExpression = getParsedExpression(expression.getOperands().get(0));
+      return expression.getOperator().equals(FunctionOperator.LENGTH)
+          ? String.format("ARRAY_LENGTH( %s, 1 )", parsedExpression)
```

Comment:
Can we add a comment explaining why the second argument is 1?
Even better would be define a constant called `DIMENSION` and add it through formatting.

Reply:
Done.